### DISCUSSION TITLE
query parameters for dynamic config

### DISCUSSION THREAD

**Discussion by hyusetiawan** [nextauthjs/next-auth#3776] [Help]
### Question 💬

As shown here: https://github.com/nextauthjs/next-auth/issues/2688
I can wrap NextAuth function, which gives me a chance to dynamically change the config via GET parameters. The end goal for me is to do login through Shopify that is non-standard, where the URL requires the shop name like so:

```
 authorization: {
          url: `https://${shopName}.myshopify.com/admin/oauth/authorize`,
          params: { grant_type: 'authorization_code', scope: 'read_customers' },
        },
```

unfortunately, NextAuth triggers a redirect which makes me lose the GET parameter. Also, is this the recommended way of doing it? I looked into `signIn()` and I do not think I can override the URL through that function? If so, I think that would be more ideal.

### How to reproduce ☕️

```
export default async function auth(req: NextApiRequest, res: NextApiResponse) {
  const { shopify } = req.query
  console.log('REQ: ', req.query)
  return await NextAuth({
...
```

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by prabureddy** [comment]
Added Shopify Provided - PR:- https://github.com/nextauthjs/next-auth/pull/3964
Changes are welcome
